src/puppetlabs/puppetdb/config.clj

```diff
@@ -375,3 +377,23 @@
       validate-vardir
       convert-config
       configure-catalog-debugging))
+
+(defprotocol DefaultedConfig
+  (get-config [this])
+  (get-in-config [this ks] [this ks default]))
```

Comment:
Is the second overload used anywhere? It seems contrary to the idea of a defaulted config service.